pkg/services/searchV2/auth.go

```diff
@@ -30,22 +29,26 @@ type dashIdQueryResult struct {
 	UID string `xorm:"uid"`
 }
 
-func (a *simpleSQLAuthService) getDashboardTableAuthFilter(user *user.SignedInUser) searchstore.FilterWhere {
-	if a.ac.IsDisabled() {
-		return permissions.DashboardPermissionFilter{
-			OrgRole:         user.OrgRole,
-			OrgId:           user.OrgID,
-			Dialect:         a.sql.GetDialect(),
-			UserId:          user.UserID,
-			PermissionLevel: dashboards.PERMISSION_VIEW,
-		}
+func (a *simpleAuthService) GetDashboardReadFilter(user *user.SignedInUser) (ResourceFilter, error) {
```

Comment:
@IevaVasiljeva just curious, is this where the major v2 improvements have been done?
Or is it the checker that is the improvement

Wanted to see more of the before/after changes. Assuming it would be here

Reply:
So the previous behaviour was as follows: query the DB for all the dashboards that the user has permissions on and then check dashboard UID against this list.

The new behaviour looks at the user permission list (which is in memory, as the user permissions should already have been retrieved by this point), and constructs a list of dashboard/folder UIDs that the user has access to. We then check permissions against this list.

So the core logic changes are here and in `checker.go`.